server/query/query.h

```diff
@@ -71,6 +71,8 @@ class Query {
            _logical_plan->is(axiom::logical_plan::NodeKind::kTableWrite);
   }
 
+  bool IsDataQuery() const { return static_cast<bool>(_logical_plan); }
```

Comment:
NIT: I know codestyle isn't clear about this, but it's ok to write `!=/== nullptr` or `has_value()` if alternative is `static_cast<bool>` or `!!expr`

I think main point of current codestyle that we want to use explicit operator bool when it's possible to do this implicit (inside if condition for an example).
`shared_ptr` or `optional`, both has `explicit operator bool`, the only exception is raw pointer, and we want to interpret this raw pointer as smart pointer, we don't want difference in how code written for raw pointers compared to smart pointers (except smart pointers have default ctor)

```suggestion
  bool IsDataQuery() const { return _logical_plan != nullptr; }
```